src/Network/Minio/Data.hs

```diff
@@ -289,19 +289,28 @@ data RequestInfo = RequestInfo {
   , riQueryParams :: Query
   , riHeaders :: [Header]
   , riPayload :: Payload
-  , riPayloadHash :: ByteString
+  , riPayloadHash :: Maybe ByteString
   , riRegion :: Maybe Region
   , riNeedsLocation :: Bool
   }
 
 instance Default RequestInfo where
-  def = RequestInfo HT.methodGet def def def def def "" def True
+  def = RequestInfo HT.methodGet def def def def def Nothing def True
 
 getPathFromRI :: RequestInfo -> ByteString
-getPathFromRI ri = B.concat parts
-  where
-    objPart = maybe [] (\o -> ["/", encodeUtf8 o]) $ riObject ri
-    parts = maybe ["/"] (\b -> "/" : encodeUtf8 b : objPart) $ riBucket ri
+getPathFromRI ri =
+  let
+    b = riBucket ri
+    o = riObject ri
+  in
+    B.concat $ ["/",
+                B.intercalate "/" $
+                map toS $
+                catMaybes $
+                b : bool [] [o] (isJust b)
```

Comment:
`catMaybes [Just "bucket", Nothing]` gives `["bucket"]`
We can just have `catMaybes (b:o)`.

Reply:
This can be further simplified as,
```haskell
B.concat ["/", B.intercalate "/" $ toS <$> catMaybes [b,o] ]
```
This simplification follows from replacing `map` with `fmap` for `[a]`. We are using `<$>`, infix `fmap`, for brevity.